staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -111,6 +108,10 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope RequestSco
 			}
 		}
 
+		requestReceived := request.TimeStampFrom(ctx)
+		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
+		timeout := parseTimeout(req.URL.Query().Get("timeout"), scope.RequestTimeout, requestReceived)
```

Comment:
there are many timeouts involved now. This is getting confusing.

Reply:
Before going deeper into the code:
- why do we need two timeouts?
- which one wins?
- does it make a difference in the returned `Status` depending on which one wins the race (if both are equal we get a race)?

Reply:
req.URL.Query().Get("timeout") is the one used before.